### PR TITLE
[DOC-3385] Update templates doc

### PR DESCRIPTION
**Pull request by brian-f-harness**
Update templates doc
- apply standards
- add numbered steps
- reorg left nav

For reviewers, previews are available:

[Templates overview](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/template/)
[Best practices and guidelines for templates](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/templates-best-practices/)
[Use a template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/use-a-template/)
[Create a pipeline template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/create-pipeline-template/)
[Create a stage template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/add-a-stage-template/)
[Create a remote stage template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/create-a-remote-stage-template/)
[Create a step template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/run-step-template-quickstart/)
[Create a remote step template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/create-a-remote-step-template/)
[Create a remote pipeline template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/create-a-remote-pipeline-template/)
[Create a secret manager template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/create-a-secret-manager-template/)
[Create an HTTP step template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/harness-template-library/)
[Create a step group template](https://64adb9f22cec7936d7640f30--harness-developer.netlify.app/docs/platform/templates/create-a-stepgroup-template/)

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/13_Templates/create-a-remote-pipeline-template.md [hunk 1]
```diff
@@ -1,139 +1,151 @@
...
 
-![](./static/create-a-remote-pipeline-template-24.png)  
+To create a remote pipeline template, do the following:
 
-For GitHub, the token must have the following scopes:
-
-![](./static/create-a-remote-pipeline-template-25.png)
+1. In your Harness Account, go to your project.
+2. In **Project SETUP**, select **Templates**.
+3. Selectd **New Template**, and then select **Pipeline**. The **Create New Pipeline Template** settings appear.
```
Comment (douglas-j-bothwell) [inline docs/platform/13_Templates/create-a-remote-pipeline-template.md:69]:
```suggestion
3. Select **New Template**, and then select **Pipeline**. The **Create New Pipeline Template** settings appear.
```

File: docs/platform/13_Templates/create-a-secret-manager-template.md [hunk 2]
```diff
@@ -1,115 +1,124 @@
...
-
-The following table shows what it means to add Templates at different scopes or hierarchies:
+### Secret manager template scope
 
+You can add secret manager templates at any [scope](../4_Role-Based-Access-Control/1-rbac-in-harness.md) in Harness.
 
+The following table shows what it means to add templates at different scopes or hierarchies:
 
 |  |  |
 | --- | --- |
```
Comment (douglas-j-bothwell) [inline docs/platform/13_Templates/create-a-secret-manager-template.md:50]: Consider including the headings in the first row, like this. Currently the headings are in the second row.
| **Scope**     | **When to add Templates?**  |
| -------- | ------- |

File: docs/platform/13_Templates/create-a-stepgroup-template.md [hunk 3]
```diff
@@ -143,29 +146,29 @@ Here is an example of a step group template YAML.
...
+If you do not use step group conditional execution settings, then the stage's conditional execution settings are applied.
 
 ## Add a failure strategy
 
 A step group can have its own **Failure Strategy** separate from the failure strategy for the stage.
 
 The failure strategy can execute the rollback steps for the step group.
 
-For more information, see [Step Failure Strategy Settings](../8_Pipelines/w_pipeline-steps-reference/step-failure-strategy-settings.md)
+For more information, go to [Step and stage failure strategy references](../8_Pipelines/w_pipeline-steps-reference/step-failure-strategy-settings.md)
```
Comment (douglas-j-bothwell) [inline docs/platform/13_Templates/create-a-stepgroup-template.md:167]:
```suggestion
For more information, go to [Step and stage failure strategy references](../8_Pipelines/w_pipeline-steps-reference/step-failure-strategy-settings.md).
```

**Review by douglas-j-bothwell — DISMISSED** [review]
A few typos and minor sugs but otherwise looks good!

File: docs/platform/13_Templates/use-a-template.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ In **Details**, you can see the following details about the selected template:
 * Type
```
[the remark this answers is not in the document]
Comment (brian-f-harness) [reply docs/platform/13_Templates/use-a-template.md:41]: updated

File: docs/platform/13_Templates/create-a-secret-manager-template.md (hunk 2 above)
Comment (brian-f-harness) [reply docs/platform/13_Templates/create-a-secret-manager-template.md:50]: Nice catch. Fixed.

File: docs/platform/13_Templates/template.md [hunk 5]
```diff
@@ -1,188 +1,189 @@
...
+You can add the following types of templates to your Harness account/org/project:
 
 * Step
 * Stage
 * Pipeline
 
-### Step Template
+### Step template
 
 To define a linear sequence of operations for a job.
```
[the remark this answers is not in the document]
Comment (brian-f-harness) [reply docs/platform/13_Templates/template.md:89]: updated

**Pull request merged**